src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyQueueCapacityAction.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.CPU;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.HEAP;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.NETWORK;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.ResourceEnum;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ModifyQueueCapacityAction implements Action {
+
+  public static final String NAME = "modify_queue_capacity";
+  public static final int COOL_OFF_PERIOD_IN_SECONDS = 300;
+
+  private int currentCapacity;
+  private int desiredCapacity;
+  private ResourceEnum threadPool;
+  private NodeKey esNode;
+
+  private Map<ResourceEnum, Integer> lowerBound = new HashMap<>();
+  private Map<ResourceEnum, Integer> upperBound = new HashMap<>();
+
+  public ModifyQueueCapacityAction(NodeKey esNode, ResourceEnum threadPool, int currentCapacity, boolean increase) {
+    setBounds();
+    int STEP_SIZE = 50;
+    this.esNode = esNode;
+    this.threadPool = threadPool;
+    this.currentCapacity = currentCapacity;
+    int desiredCapacity = increase ? currentCapacity + STEP_SIZE : currentCapacity - STEP_SIZE;
```

Comment:
we only consider increase or decrease of queue resources here. Is it safe to assume that an obj will not be created at all if there is no change in values ?

Reply:
Actions are created by deciders only when they need to be triggered, and they always do something; hence the assumption.

Reply:
These are good points. We should add them to the class level javadoc